### PR TITLE
USD Contribution Workflow: Show/hide attributes per instance based on status of other toggles

### PR DESCRIPTION
**Pull request by BigRoy**
## Changelog Description

Show/hide attributes per instance for the USD contribution workflow based on status of other toggles in the Publisher UI.

## Additional info

Replaces PR https://github.com/ynput/ayon-core/pull/943 which I couldn't seem to retarget to `develop` anymore

## Testing notes:

1. Create USD instance in Maya, Houdini or Blender
2. The attributes should update on toggling on/off the toggles for Contribution Worfklow.
3. Extra test: USD Look publishing in Houdini should still work.

See screenshots:

image
image
image

### CODE REVIEW

File: client/ayon_core/plugins/publish/extract_usd_layer_contributions.py [hunk 1]
```diff
@@ -458,7 +458,22 @@ def get_or_create_instance(self,
...
 
     @classmethod
-    def get_attribute_defs(cls):
+    def get_attr_defs_for_instance(cls, create_context, instance):
+        # Filtering of instance, if needed, can be customized
+        if not cls.instance_matches_plugin_families(instance):
+            return []
+
+        # Attributes logic
+        disabled = False
```
Comment (iLLiCiTiT) [inline client/ayon_core/plugins/publish/extract_usd_layer_contributions.py:467]: This variable should be `hidden`, or better `visible`.

Comment (BigRoy) [reply client/ayon_core/plugins/publish/extract_usd_layer_contributions.py:467]: Fixed with https://github.com/ynput/ayon-core/pull/960/commits/d797b78fcaa1bdbc3836036608744318a36f1045

**Comment by MustafaJafar** [comment]
It works in Houdini but, there's a weird error print in the console. 
This error was printed 25 times when I refresh the publisher .
```
ERROR:CreateContext:Failed to get attribute definitions from plugin 'CollectUSDLayerContributionsHoudiniLook'.
Traceback (most recent call last):
  File "E:\Ynput\ayon-core\client\ayon_core\pipeline\create\context.py", line 1254, in bulk_add_instances
    attr_defs = plugin.get_attr_defs_for_instance(
  File "E:\Ynput\ayon-core\client\ayon_core\plugins\publish\extract_usd_layer_contributions.py", line 606, in get_attr_defs_for_instance
    layer_def = next(d for d in defs if d.key == "contribution_layer")
StopIteration
```

**Review by MustafaJafar — APPROVED** [review]
Tested in Houdini, Maya and Blender. It works. 
The issue mentioned in my previous comments shows only in Houdini.

**Comment by MustafaJafar** [comment]
> It works in Houdini but, there's a weird error print in the console. This error was printed 25 times when I refresh the publisher .
> 
> ```python
> ERROR:CreateContext:Failed to get attribute definitions from plugin 'CollectUSDLayerContributionsHoudiniLook'.
> Traceback (most recent call last):
>   File "E:\Ynput\ayon-core\client\ayon_core\pipeline\create\context.py", line 1254, in bulk_add_instances
>     attr_defs = plugin.get_attr_defs_for_instance(
>   File "E:\Ynput\ayon-core\client\ayon_core\plugins\publish\extract_usd_layer_contributions.py", line 606, in get_attr_defs_for_instance
>     layer_def = next(d for d in defs if d.key == "contribution_layer")
> StopIteration
> ```

I furtherly checked this, firstly I added a print statement inside the [CollectUSDLayerContributionsHoudiniLook.get_attr_defs_for_instance](https://github.com/ynput/ayon-core/blob/d2ee4167ae0151908eda349d582663bf193efdd9/client/ayon_core/plugins/publish/extract_usd_layer_contributions.py#L559)
and the print didn't show up for those 25 times I was talking about. 
therefore, the `"contribution_layer"` doesn't exist all the times resulting the next to fail to find any values which is a lot similar to 
```python
next(i for i in range(4) if i > 5)
```

**Comment by BigRoy** [comment]
Thanks so much for catching that @MustafaJafar. Fixed with https://github.com/ynput/ayon-core/pull/960/commits/11bb657d35921c42759339937ee8b82230235e98

**Comment by MustafaJafar** [comment]
> Thanks so much for catching that @MustafaJafar. Fixed with [11bb657](https://github.com/ynput/ayon-core/commit/11bb657d35921c42759339937ee8b82230235e98)

Cool The error has disappeared.

**Comment by BigRoy** [comment]
Please review @iLLiCiTiT and merge if happy with the changes.

**Review by iLLiCiTiT — APPROVED** [review]
Codewise looks ok. If it works happy to merge.

**Pull request merged**